### DISCUSSION TITLE
make auto-scroll optional

### DISCUSSION THREAD

**Discussion by vesper8** [spatie/ray#362] [Ideas]
Can we please make the auto-scroll optional so that if you're for whatever reason getting a stream of new entries coming in you can scroll up to an error without having it auto-scroll to the bottom/top every time there's a new entry

Also I noticed the Ray window on Mac is "always on top" and this does not appear to be configurable. I have only limited screen space and I don't want Ray to always be on top of other windows. I understand this may be desirable for some but please make it configurable

**Comment by freekmurze** [comment]
I like this idea!

**Comment by AdrianMrn** [comment]
@vesper8 I'll have a look at the scroll behaviour! The "always on top" behaviour can be toggled using the "float" button on the top-right of the application:

![image](https://user-images.githubusercontent.com/12762044/110967299-83682a00-8356-11eb-954f-11b2adc68851.png)

**Reply by vesper8** [reply]
I was wondering what that icon was.. but now I see that hovering over it does mention the float thing

**Reply by vesper8** [reply]
@AdrianMrn It would also be very nifty if you could toggle an option that would bring the Ray window to the front when it receives a new entry

**Reply by vesper8** [reply]
@AdrianMrn did you end up removing this toggle button? I don't see it anymore in 1.19.0

**Comment by Tarpsvo** [comment]
I am looking for this exact feature.

The most common UX in this scenario is like in any messenger application - when you've scrolled away from the bottom it will not move your scroll position when new entries are added. Instead, the application usually displays a scroll to bottom button with the number of new entries near it so you know how many new things have been logged that you have not yet seen.

Example:
![image](https://user-images.githubusercontent.com/2018660/136358126-c8321c0d-33aa-4370-b1bd-d0377fe7b8f8.png)

**Reply by AdrianMrn** [reply]
@Tarpsvo The scroll-on-log toggle is already a configuration option: 
![image](https://user-images.githubusercontent.com/12762044/136359235-628907b7-f09f-47b5-a6eb-1aca395c953a.png)

Great idea about the scroll button with the counter, I'll add that!

**Reply by Tarpsvo** [reply]
Thanks! I did not know that. :) Toggled it off for now.

**Comment by vesper8** [comment]
@AdrianMrn @freekmurze Thanks for adding this configuration open.

I actually find myself toggling it on and off pretty frequently now depending on what I'm doing, it would be nice if I could easily toggle it with an icon in the taskbar, similar to the 'always on top' toggle button.. which.. appears to have gone missing now.. actually

**Reply by AdrianMrn** [reply]
@vesper8 We don't want to add any other buttons to the taskbar anymore for now. The always-on-top button has also moved to a setting in the preferences (you can however configure a global shortcut to toggle this).

We'll add something like this when scroll-on-log is disabled, would this be a suitable alternative solution for your issue?
![image](https://user-images.githubusercontent.com/12762044/149367563-8798f218-7e1c-4a29-9392-ee89c27b30e8.png)

**Reply by vesper8** [reply]
@AdrianMrn  Being able to toggle the auto-scroll on and off with a keyboard shortcut would be useful for sure. But I would have preferred to have a visual cue to let me know whether it's on or off, whether that's a toggle or a little red/green light, it would be useful. I understand that you don't want to clutter the taskbar with buttons, but personally I would want these buttons.. so if it was possible to enable them via the configuration that would be even better. But as usual.. I'll take what I can get : ) Cheers!
